### PR TITLE
fix: "New chat" resets thread

### PR DESCRIPTION
**Pull request by kgilpin**

### CODE REVIEW

File: packages/components/src/components/chat/Chat.vue [hunk 1]
```diff
@@ -115,32 +149,30 @@ export default {
...
-        message: content || '',
-        sentiment: 0,
-      };
-      this.messages.push(message);
-
+    // Creates a new message, from the user or from the AI.
+    addUserMessage(content: string) {
+      this.messages.push(new UserMessage(content));
+      this.scrollToBottom();
+    },
```
Comment (dustinbyrne) [inline packages/components/src/components/chat/Chat.vue:172]: Is this comment accurate? It looks like this is only for user messages.

Comment (kgilpin) [reply packages/components/src/components/chat/Chat.vue:172]: Yeah, it looks like it's only for User messages.

File: packages/components/tests/unit/chat/ChatHelp.spec.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      expect(wrapper.find(`.messages`).text()).toContain('Begin by installing the AppMap agent');
+
+      await waitFor(
+        `.messages should contain " then configure your project per the instructions"`,
+        () =>
+          wrapper
+            .find(`.messages`)
+            .text()
+            .includes(' then configure your project per the instructions')
+      );
```
Comment (dustinbyrne) [inline packages/components/tests/unit/chat/ChatHelp.spec.js:59]: `waitFor` doesn't look to be necessary. These elements should all be in an expected state the first time this assertion is made.

File: packages/components/tests/unit/chat/ChatHelp.spec.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        ]);
+
+        expect(wrapper.find('.status-unauthorized').exists()).toBe(false);
+        expect(wrapper.find('.messages .message').exists()).toBe(true);
+        expect(wrapper.find(`.messages`).text()).toContain('Begin by installing the AppMap agent');
+
+        // NOTE: Ideally this would have different formatting than a regular server message.
+        await waitFor(`.messages should contain "Server error"`, () =>
+          wrapper.find(`.messages`).text().includes('Server error')
+        );
```
Comment (dustinbyrne) [inline packages/components/tests/unit/chat/ChatHelp.spec.js:94]: Same thing with this `waitFor`

**Review by dustinbyrne — APPROVED** [review]

File: packages/components/tests/unit/chat/ChatHelp.spec.js (hunk 2 above)
Comment (kgilpin) [reply packages/components/tests/unit/chat/ChatHelp.spec.js:59]: They aren't though.

**Pull request merged**